libraries/Microsoft.Bot.Builder.Core.Extensions/CatchExceptionMiddleware.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class CatchExceptionMiddleware<T> : IMiddleware where T : Exception
```

Comment:
>CatchExceptionMiddleware [](start = 17, length = 24)

Description of use and benefit?

Reply:
This one is pretty useful and cool

---
In reply to: [174949931](https://github.com/Microsoft/botbuilder-dotnet/pull/283#discussion_r174949931) [](ancestors = 174949931)

Reply:
Description added.